### PR TITLE
Added Russian 'functions' lesson

### PR DESCRIPTION
**Pull request by brain-geek**

### CODE REVIEW

File: ru/lessons/basics/functions.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+В Elixir, как и в многих других функциональных языках, функции являются полноценными объектами. В этом уроке мы рассмотрим типы функций в Elixir, чем они отличаются и как их использовать.
+
+## Содержание
+
+- [Анонимные функции](#section-1)
+  - [Краткий синтаксис](#section-2)
+- [Сопоставление с образцом](#section-3)
+- [Именованные функции](#section-4)
+  - [Закрытые функции](#section-5)
+  - [Органичители](#section-6)
```
Comment (gaynetdinov) [inline ru/lessons/basics/functions.md]: A typo: `Ограничители`

Comment (brain-geek) [inline ru/lessons/basics/functions.md]: Fixed.

**Comment by brain-geek** [comment]
Hey, @gaynetdinov , can you look through this PR once again?

File: ru/lessons/basics/functions.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+  end
+
+  defp phrase("en"), do: "Hello, "
+  defp phrase("es"), do: "Hola, "
+end
+
+** (CompileError) def hello/2 has default values and multiple clauses, define a function head with the defaults
+```
+
+Elixir не поддерживает аргументы по умолчанию при наличии нескольких подходящих функций, ведь это может запутать. Для решения этой проблемы мы добавляем определение функции с аргументами по умолчанию:
```
Comment (gaynetdinov) [inline ru/lessons/basics/functions.md]: I don't feel comfortable with the phrase `, ведь это может запутать.` However, I don't have better translation either. 
Other than that, this PR looks good to me.

Comment (brain-geek) [inline ru/lessons/basics/functions.md]: Well... I don't like it too. This part can be thrown out, but then we're moving more and more to literary translation.

File: ru/lessons/basics/functions.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+```elixir
+iex> sum = fn (a, b) -> a + b end
+iex> sum.(2, 3)
+5
+```
+
+### Краткий синтаксис
+
+Использование анонимных функций - настолько часто используемая возможность языка, что для нее есть специальное сокращение:
```
Comment (tetiana12345678) [inline ru/lessons/basics/functions.md]: hello. sorry for being noisy. Just going through this PR. Do we need `-` here?

File: ru/lessons/basics/functions.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+"Hello, Sean"
+
+iex> Greeter.phrase
+** (UndefinedFunctionError) undefined function: Greeter.phrase/0
+    Greeter.phrase()
+```
+
+### Ограничители
+
+Мы уже затрагивали ограничители в главе [Управляющие конструкции](../control-structures.md), теперь же рассмотрим их применение в именованных функциях. Ограничители проверяются только после того как только Elixir сопоставил функцию.
```
Comment (tetiana12345678) [inline ru/lessons/basics/functions.md]: Maybe `только после того как`. Using `только` 2 times in the sentence makes it a bit confusing :stuck_out_tongue:

File: ru/lessons/basics/functions.md [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+
+В следующем примере у нас есть две функции с одинаковыми сигнатурами. Мы используем ограничители для определения какую из них использовать на основе типа аргумента:
```
Comment (tetiana12345678) [inline ru/lessons/basics/functions.md]: I haven't read programming book in Russian for ages, so I am not best person to judge. Is `сигнатуры` a common term for `signatures` is Russian language?

File: ru/lessons/basics/functions.md (hunk 3 above)
Comment (brain-geek) [inline ru/lessons/basics/functions.md]: As I understand, it is.

File: ru/lessons/basics/functions.md (hunk 5 above)
Comment (brain-geek) [inline ru/lessons/basics/functions.md]: In this context (function definitions) - yes. https://ru.wikipedia.org/wiki/API#.D0.A1.D0.B8.D0.B3.D0.BD.D0.B0.D1.82.D1.83.D1.80.D0.B0_.D1.84.D1.83.D0.BD.D0.BA.D1.86.D0.B8.D0.B8

> Сигнатура функции — часть общего объявления функции, позволяющая средствам трансляции идентифицировать функцию среди других.

**Comment by brain-geek** [comment]
@tetiana12345678 Thanks for the feedback! I've fixed everything you noted.

**Comment by tetiana12345678** [comment]
@brain-geek looks good to me :+1: I have started to work on translation for ecto in specifics. Wasn't sure what you are working on, so, thought it's the best way to avoid clushing. My pr is on the way :)

**Comment by brain-geek** [comment]
Hey, @pjhampton or @doomspork , can you merge this? Looks like it's fine.

**Comment by pjhampton** [comment]
Done @brain-geek! :+1:

**Comment by brain-geek** [comment]
Thanks!

**Comment by doomspork** [comment]
@brain-geek I wanted to let you know I've added you as a project collaborator so you can merge in RU translations.  Please refrain from pushing to master directly, I'd like everything to go through PR so there's a chance for other contributors to offer feedback.  :+1:

**Comment by brain-geek** [comment]
Thank you for trust, @doomspork !

**Pull request merged**